specs/hotel_spec.rb

```diff
@@ -0,0 +1,247 @@
+      @test_ob.all_block_reservations[0].price.must_equal 150
+    end
+
+    it "Raises UnavailableBlockError if there are no blocks available" do
+      proc { @test_ob.block_reservation(@room, @check_in, @check_out) }.must_raise BookingSystem::Hotel::UnavailableBlockError
+    end
+
+    it "Raises UnavailableBlockError if there are no blocks that match date range" do
+      @test_ob.block_off_a_block(3, @check_in, @check_out)
+      proc { @test_ob.block_reservation(@room, Date.new(2017,10,9), Date.new(2017,10,10)) }.must_raise BookingSystem::Hotel::UnavailableBlockError
+    end
+
+    it "Raises UnavailableRoomError if the room requested is not available" do
+      @test_ob.block_off_a_block(3, @check_in, @check_out)
+      proc { @test_ob.block_reservation(20, @check_in, @check_out) }.must_raise BookingSystem::Hotel::UnavailableRoomError
+    end
+
+    it "Adds a new block reservation to @all_block_reservations" do
+      @test_ob.block_off_a_block(4, @check_in, @check_out)
+      @test_ob.block_reservation(2, @check_in, @check_out)
+      @test_ob.all_block_reservations.length.must_equal 1
+    end
+
+    it "Returns an instance of BlockReservation" do
+      @test_ob.block_off_a_block(4, @check_in, @check_out)
+      @test_ob.block_reservation(2, @check_in, @check_out).must_be_instance_of BookingSystem::BlockReservation
+    end
+  end
+
+  describe "block_off_a_block method" do
+    it "Can be called" do
+      @test_ob.must_respond_to :block_off_a_block
+    end
+
+    it "Raises UnavailableRoomError if number of rooms requested exceeds available rooms" do
+      4.times do
+        @test_ob.block_off_a_block(5, @check_in, @check_out)
+      end
+
+      proc { @test_ob.block_off_a_block(3, @check_in, @check_out) }.must_raise BookingSystem::Hotel::UnavailableBlockError
+    end
+
+    it "Returns an instance of Block class" do
+      @test_ob.block_off_a_block(3, @check_in, @check_out).must_be_instance_of BookingSystem::Block
+    end
+
+    it "Increases the length of the @all_blocks by 1" do
+      @test_ob.block_off_a_block(3, @check_in, @check_out)
+      @test_ob.all_blocks.length.must_equal 1
+    end
+  end
+
+  describe "reservations_for_specific_date" do
+    it "Can be called" do
+      @test_ob.must_respond_to :reservations_for_specific_date
+    end
+
+    it "Returns an array of Reservation objects" do
+      @test_ob.make_single_reservation(@room, @check_in, @check_out)
+
+      @test_ob.make_single_reservation(2, Date.new(2017,9,11), Date.new(2017,9,15))
+      @test_ob.block_off_a_block(4, @check_in, @check_out)
+      @test_ob.block_reservation(3, @check_in, @check_out)
+
+      @test_ob.reservations_for_specific_date(Date.new(2017,9,10)).must_be_instance_of Array
+
+      @test_ob.reservations_for_specific_date(Date.new(2017,9,10))[0].must_be_instance_of BookingSystem::Reservation
+
+      @test_ob.reservations_for_specific_date(Date.new(2017,9,11)).length.must_equal 3
+    end
```

Comment:
Good start. Other cases I'd be interested in:
- What if there are no reservations? (it should return an empty list)
- Do reservations that don't overlap the date range affect the list? (they shouldn't)
  - Check the boundaries of what counts as an overlap. What about a reservation that ends on the checkin date, or starts on the checkout date?